content/en/docs/concepts/workloads/pods/pod-lifecycle.md

```diff
@@ -378,6 +378,47 @@ are affected by the same inconsistencies as the regular restart policy. Kubelet
 runtime garbage collection, intermitted connectivity issues with the control plane may cause the state
 loss and containers may be re-run even when you expect a container not to be restarted.
 
+#### Restart All Containers {#restart-all-containers}
+
+{{< feature-state feature_gate_name="RestartAllContainersOnContainerExits" state="alpha" >}}
+
+If your cluster has the feature gate `RestartAllContainersOnContainerExits` enabled, you can specify
+`RestartAllContainers` as an action in `restartPolicyRules`. When a container's exit matches a rule with this action, the entire Pod is terminated and restarted in-place.
+
+This "in-place" restart offers a more efficient way to reset a Pod's state compared to full deletion and recreation. This is especially valuable for workloads where rescheduling is costly, such as batch jobs or AI/ML training tasks.
+
+##### How in-place Pod restarts work
+
+When a `RestartAllContainers` action is triggered, the kubelet performs the following steps:
+
+1.  **Fast Termination**: All running containers in the Pod are terminated. The configured `terminationGracePeriodSeconds` is not respected, and any configured `preStop` hooks are not executed. This ensures a swift shutdown.
```

Comment:
Please wrap the long line at the semantic boundaries.
Please avoid using 1, 2, 3, 4 as the bulletin numbers, use 1, 1, 1 ... instead.